core/templates/dev/head/pages/exploration_editor/RouterServices.js

```diff
@@ -44,14 +44,20 @@ oppia.factory('routerService', [
       return $location.path();
     }, function(newPath, oldPath) {
       if (newPath === '') {
-        $location.path(oldPath);
+        if (oldPath === '/') {
+          $window.history.go(-1);
+        } else {
+          $location.path(oldPath);
+        }
         return;
       }
-
       if (!oldPath) {
         // This can happen when clicking on links whose href is "#".
         return;
       }
+      if (newPath === '/' && oldPath.includes('/gui/')) {
```

Comment:
Hmm. Instead of carrying context from the dashboard around and trying to work around it like this, would it be better to, in the dashboard, do some kind of replaceState() from the history stack if there's an automatic redirect? Then this page does not have to know about the user's navigation history, or special-case on it. I think that would be cleaner and easier to understand.